scipy/sparse/linalg/__init__.py

```diff
@@ -117,15 +117,18 @@
 
 """
 
-from .isolve import *
-from .dsolve import *
-from .interface import *
-from .eigen import *
-from .matfuncs import *
+from ._isolve import *
+from ._dsolve import *
+from ._interface import *
+from ._eigen import *
+from ._matfuncs import *
 from ._onenormest import *
 from ._norm import *
 from ._expm_multiply import *
 
+# Deprecated namespaces, to be removed in v2.0.0
+from . import isolve, dsolve, interface, eigen, matfuncs
```

Comment:
I tested this in some detail, because there's multiple chained namespaces. It all seems to work as expected. For example:
```python
>>> from scipy.sparse.linalg import eigen
>>> eigen.arpack.ArpackError
<ipython-input-5-5af8690eb2ea>:1: DeprecationWarning: Please use `arpack` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.eigen` namespace is deprecated.
  eigen.arpack.ArpackError
<class 'scipy.sparse.linalg._eigen.arpack.arpack.ArpackError'>
```